module/framework.js

```diff
@@ -449,19 +473,41 @@ async function _getCategories({cache=true}={}) {
     id: CATEGORY_ID.uncategorized,
     title: game.i18n.localize('WA.CategoryUncategorized'),
     position: 9e9,
+    copyForSort: [],
     children : [],
     parent: root,
     isUncategorized: true
   };
   categories.set(uncategorized.id, uncategorized);
 
-  // Retrieve categories from the World Anvil API
+  // Retrieve categories from the World Anvil API (build map)
   const request = await anvil.getCategories();
   for ( let c of (request?.categories || []) ) {
+    categories.set(c.id, c);
+    c.copyForSort = c.children?.categories ?? [];
     c.children = [];
     c.folder = undefined;
-    categories.set(c.id, c);
   }
+  // Append children 
+  for( let c of (request?.categories || []) ) {
+    const parentId = c.parentCategory?.id ?? CATEGORY_ID.root;
+    const parent = categories.get(parentId);
+    c.parent = parent;
+    parent.children.push(c);
+  }
+  // Sort children
+  for( let c of categories.values() ) {
+
+    c.children.sort( (a,b) => {
+      const indexA = c.copyForSort.findIndex( cc => cc.id === a.id );
```

Comment:
Can you help me understand the role that `copyForSort` is playing here? If we are sorting based on the indices of the un-sorted array, why aren't we just using the un-sorted array directly?

Reply:
We only want one instance of each category.
What we retrieve from our API call are copies.

So we wait for the contruction of each instance and rebuild its childrens by taking into account the ordrer given in the copy.
Only the id of each element is important in this list.